src/main/java/io/vertx5/core/net/NetSocket.java

```diff
@@ -104,41 +105,65 @@ public NetSocket closeHandler(Handler<Void> handler) {
     return this;
   }
 
-  final ChannelHandler handler = new ChannelHandler() {
+  static class ChannelHandlerImpl implements ChannelHandler {
+
+    private final NetSocket netSocket;
+
+    public ChannelHandlerImpl(NetSocket netSocket) {
+      this.netSocket = netSocket;
+    }
+
     @Override
-    public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
-      channelHandlerContext = ctx;
+    public void channelActive(ChannelHandlerContext ctx) {
+      netSocket.writable = ctx.channel().isWritable();
     }
+
     @Override
-    public void handlerRemoved(ChannelHandlerContext ctx) throws Exception {
-      channelHandlerContext = null;
+    public void channelWritabilityChanged(ChannelHandlerContext ctx) throws Exception {
+      boolean writable = !netSocket.writable;
+      netSocket.writable = writable;
+      if (writable && netSocket.needsFlush) {
```

Comment:
it shouldn't load the writability from ctx.channel().isWritable?

Reply:
I'm assuming that the event is triggered when the value changes, so using writable = !writable seems correct